php/class-post-type.php

```diff
@@ -549,4 +548,37 @@ public function filter_user_has_cap( $allcaps, $caps ) {
 
 		return $allcaps;
 	}
+
+	/**
+	 * Publish snapshot changes when snapshot post is being published.
+	 *
+	 * @param int      $post_id Post id.
+	 * @param \WP_Post $post Post object.
+	 */
+	public function publish_snapshot( $post_id, $post ) {
+		if ( ! is_a( $this->snapshot_manager->customize_manager, '\WP_Customize_Manager' ) ) {
+			require_once( ABSPATH . WPINC . '/class-wp-customize-manager.php' );
+			$this->snapshot_manager->customize_manager = new \WP_Customize_Manager();
+		}
+		$snapshot_content = json_decode( $post->post_content, true );
+		if ( json_last_error() || ! is_array( $snapshot_content ) ) {
+			return;
+		}
+		$snapshot_values = array_filter(
+			wp_list_pluck( $snapshot_content, 'value' ),
+			function( $value ) {
+				return ! is_null( $value );
+			}
+		);
```

Comment:
You'll also want to do this here, to ensure all of the expected actions are done as found in `WP_Customize_Manager::save()`:

```php
if ( method_exists( $this->snapshot_manager->customize_manager, 'validate_setting_values' ) ) {
    do_action( 'customize_save_validation_before', $this->snapshot_manager->customize_manager );
}
```